src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5838,122 +5850,136 @@ cJSON *wm_vuldet_get_cvss(const char *scoring_vector) {
-            } else if (impact == 'C') {
-                cJSON_AddStringToObject(cvss_json, "integrity_impact", "complete");
-            } else if (impact == 'L') {
-                cJSON_AddStringToObject(cvss_json, "integrity_impact", "low");
-            } else if (impact == 'H') {
-                cJSON_AddStringToObject(cvss_json, "integrity_impact", "high");
-            }
-        }
-        // Availability
-        if (vector_it = strstr(scoring_vector, cvss_availability), vector_it) {
-            impact = *(vector_it + strlen(cvss_availability));
-            if (impact == 'N') {
-                cJSON_AddStringToObject(cvss_json, "availability", "none");
-            } else if (impact == 'P') {
-                cJSON_AddStringToObject(cvss_json, "availability", "partial ");
-            } else if (impact == 'C') {
-                cJSON_AddStringToObject(cvss_json, "availability", "complete");
-            } else if (impact == 'L') {
-                cJSON_AddStringToObject(cvss_json, "availability", "low");
-            } else if (impact == 'H') {
-                cJSON_AddStringToObject(cvss_json, "availability", "high");
-            }
-        }
-        // Privileges required
-        if (vector_it = strstr(scoring_vector, cvss_privileges_required), vector_it) {
-            impact = *(vector_it + strlen(cvss_privileges_required));
-            if (impact == 'N') {
-                cJSON_AddStringToObject(cvss_json, "privileges_required", "none");
-            } else if (impact == 'P') {
-                cJSON_AddStringToObject(cvss_json, "privileges_required", "partial ");
-            } else if (impact == 'C') {
-                cJSON_AddStringToObject(cvss_json, "privileges_required", "complete");
-            } else if (impact == 'L') {
-                cJSON_AddStringToObject(cvss_json, "privileges_required", "low");
-            } else if (impact == 'H') {
-                cJSON_AddStringToObject(cvss_json, "privileges_required", "high");
-            }
-        }
-        // User interaction
-        if (vector_it = strstr(scoring_vector, cvss_user_interaction), vector_it) {
-            impact = *(vector_it + strlen(cvss_user_interaction));
-            if (impact == 'N') {
-                cJSON_AddStringToObject(cvss_json, "user_interaction", "none");
-            } else if (impact == 'R') {
-                cJSON_AddStringToObject(cvss_json, "user_interaction", "required ");
-            }
-        }
-        // Scope
-        if (vector_it = strstr(scoring_vector, cvss_scope), vector_it) {
-            impact = *(vector_it + strlen(cvss_scope));
-            if (impact == 'U') {
-                cJSON_AddStringToObject(cvss_json, "scope", "unchanged");
-            } else if (impact == 'C') {
-                cJSON_AddStringToObject(cvss_json, "scope", "changed ");
+        char* token = NULL;
+
+        for (token = strtok(vector, "/"); NULL != token; token = strtok(NULL, "/"))
```

Comment:
This function was always returning wrong results. The thing is that because of the comparison of very similar fields like `"AC:"` and `"C:"`, or `"UI:"` and `"I:"`, the function was returning two times`access_complexity` values and never `confidentiality_impact`, as well as two times `integrity_impact` and never `user_interaction`.